tools/filesystem/file.go

```diff
@@ -87,6 +100,31 @@ func (r *PathReader) Open() (io.ReadSeekCloser, error) {
 
 // -------------------------------------------------------------------
 
+var _ FileReader = (*BufReader)(nil)
+
+type BufReader struct {
+	Bytes []byte
+}
+
+// Open implements the [filesystem.FileReader] interface.
+func (r *BufReader) Open() (io.ReadSeekCloser, error) {
+	return NewBytesReadSeekCloser(r.Bytes), nil
+}
+
+type BytesReadSeekCloser struct {
+	*bytes.Reader
+}
+
+func NewBytesReadSeekCloser(b []byte) *BytesReadSeekCloser {
```

Comment:
Since this is currently used only inside the `Open()`, let's change it to `newBytesReadSeekCloser()` to limit the surface of the exported package members (or remove it and directly call the returned struct literal in its place).